pkg/apiserver/resthandler.go

```diff
@@ -676,6 +676,13 @@ func UpdateResource(r rest.Updater, scope RequestScope, typer runtime.ObjectType
 			}
 		}
 
+		canDelete := func() bool {
```

Comment:
> @liggitt @lavalamp done right, this would let us do:

A spot to inject a field based check would be sufficient to let us support the permission check we want without forcing in a server that doesn't want it.  That would work for me.